src/aap_eda/tasks/analytics.py

```diff
@@ -27,20 +28,30 @@
 ANALYTICS_JOB_ID = "job_gather_analytics"
 ANALYTICS_TASKS_QUEUE = "default"
 
+# Wrap the django_rq job decorator so its processing is within our retry
+# code.
+job = tasking.redis_connect_retry()(django_rq.job)
 
-def schedule_gather_analytics(queue_name: str = ANALYTICS_TASKS_QUEUE) -> None:
+
+def schedule_gather_analytics(
+    queue_name: str = ANALYTICS_TASKS_QUEUE, cancel: bool = False
+) -> None:
     if not flag_enabled("FEATURE_EDA_ANALYTICS_ENABLED"):
         return
     interval = utils.get_analytics_interval()
     logger.info(f"Schedule analytics to run in {interval} seconds")
-    with advisory_lock(ANALYTICS_SCHEDULE_JOB_ID, wait=False) as acquired:
+    with advisory_lock(ANALYTICS_SCHEDULE_JOB_ID, wait=cancel) as acquired:
         if not acquired:
             logger.debug(
                 "Another instance of schedule_gather_analytics"
                 " is already running",
             )
             return
 
+        if cancel:
```

Comment:
@bzwei What would happen if the interval is set to a low value like 2 seconds and if the gather job doesn't finish in 2 seconds will we end up killing the old running job. The next job would exhibit the same behavior and we will never get the gather task to finish successfully. Should we have a setting called
MIINIMUM_GATHER_TASK_TIME and ensure that the user cannot change the interval to anything less than MIINIMUM_GATHER_TASK_TIME

Reply:
@mkanoor A running job continues to run to finish, never be killed. I proved this by doing real tests.